spec/rspec/mocks/argument_matchers_spec.rb

```diff
@@ -381,16 +381,26 @@ def ==(other)
           a_double.random_call(:a => "a", :b => "b")
         end
 
-        it "matches against a hash submitted by reference and received by value" do
+        it "matches against a hash submitted as keyword arguments a and received as a positional argument (in both Ruby 2 and Ruby 3)" do
           opts = {:a => "a", :b => "b"}
           expect(a_double).to receive(:random_call).with(opts)
           a_double.random_call(:a => "a", :b => "b")
         end
 
-        it "matches against a hash submitted by value and received by reference" do
-          opts = {:a => "a", :b => "b"}
-          expect(a_double).to receive(:random_call).with(:a => "a", :b => "b")
-          a_double.random_call(opts)
+        if RUBY_VERSION >= "2.7"
+          it "fails to matches against a hash submitted as a positional argument and received as keyword arguments in Ruby 2.7 or later", :reset => true do
```

Comment:
My last concern is that this is a breaking change for 2.7, as 2.7 did accept hashes submitted in this fashion, it just issues a warning, WDYT about changing this to work for Ruby 3 only, or making it configurable for Ruby 2.7 for now, but we'll remove the config in RSpec 4 making this the permanent option?

I'm happy to tackle that in an after PR myself if you cannot.

Reply:
Sorry for my slow reaction! I've made the change only since 3.0, and the CI has passed. I think that making it configurable for Ruby 2.7 is reasonable, but may I leave the implementation to you? Thanks!